### PR TITLE
Fix opds format errors

### PR DESCRIPTION
**Pull request by xcffl**
There are several errors in current opds templetes.

* ~~Not using absolute URL~~
* Not specifying `kind` in `type`
* Some titles and contents are identical

This PR fixes all above. Due to string changes, I also regenerated the translation files. If that's not correct, I can remove that commit.

### CODE REVIEW

**Comment by OzzieIsaacs** [comment]
I don‘t see the benefit having absolute links, even the specification is using relative links in their examples: https://specs.opds.io/opds-1.2.html#22-navigation-feeds

**Comment by xcffl** [comment]
Yes, just the Atom spec requires absolute link. Maybe I'll remove that if it's unnecessary.

**Comment by xcffl** [comment]
Done

**Comment by xcffl** [comment]
Oh I know why the Atom spec forbids relative paths - the subscription file might be provided by someone on other domain

File: .gitignore [hunk 1]
```diff
@@ -30,3 +30,5 @@ settings.yaml
 gdrive_credentials
 client_secrets.json
 
+.vscode/settings.json
```
Comment (OzzieIsaacs) [inline .gitignore:33]: Could we please just exclude the whole subtree, I have no feeling if there will be some more files generated under certain conditions

File: cps/templates/feed.xml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<feed xmlns="http://www.w3.org/2005/Atom" xmlns:dcterms="http://purl.org/dc/terms/">
+<feed xmlns="http://www.w3.org/2005/Atom" xmlns:dc="http://purl.org/dc/terms/">
   <id>urn:uuid:2853dacf-ed79-42f5-8e8a-a7bb3d1ae6a2</id>
```
Comment (OzzieIsaacs) [inline cps/templates/feed.xml:3]: okay, conform to specification and calibre opds feed

File: cps/templates/feed.xml [hunk 3]
```diff
@@ -49,26 +49,32 @@
...
+    {% endfor %}
     {% if entry.comments[0] %}<summary>{{entry.comments[0].text|striptags}}</summary>{% endif %}
     {% if entry.has_cover %}
     <link type="image/jpeg" href="{{url_for('opds.feed_get_cover', book_id=entry.id)}}" rel="http://opds-spec.org/image"/>
     <link type="image/jpeg" href="{{url_for('opds.feed_get_cover', book_id=entry.id)}}" rel="http://opds-spec.org/image/thumbnail"/>
     {% endif %}
     {% for format in entry.data %}
     <link rel="http://opds-spec.org/acquisition" href="{{ url_for('opds.opds_download_link', book_id=entry.id, book_format=format.format|lower)}}"
-          length="{{format.uncompressed_size}}" mtime="{{entry.atom_timestamp}}" type="{{format.format|lower|mimetype}}"/>
+          length="{{format.uncompressed_size}}" type="{{format.format|lower|mimetype}}"/>
```
Comment (OzzieIsaacs) [inline cps/templates/feed.xml:77]: Calibre adds this entry to it's feed and many opds reader are more compatible to calibre than to the spec and some of them are very strict, if there is anything different form what they are expecting it fails. Last time I changed something several readers caused problems -> No change

File: cps/templates/index.xml [hunk 4]
```diff
@@ -15,101 +15,101 @@
   </author>
   <entry>
     <title>{{_('Hot Books')}}</title>
-    <link href="{{url_for('opds.feed_hot')}}" type="application/atom+xml;profile=opds-catalog"/>
+    <link href="{{url_for('opds.feed_hot')}}" type="application/atom+xml;profile=opds-catalog;kind=acquisition"/>
     <id>{{url_for('opds.feed_hot')}}</id>
```
Comment (OzzieIsaacs) [inline cps/templates/index.xml:19]: Calibre adds this entry to it's feed and many opds reader are more compatible to calibre than to the spec and some of them are very strict, if there is anything different form what they are expecting it fails. Last time I changed something several readers caused problems -> No change

File: cps/templates/index.xml [hunk 5]
```diff
@@ -15,101 +15,101 @@
...
     <updated>{{ current_time }}</updated>
     <content type="text">{{_('Popular publications from this catalog based on Downloads.')}}</content>
   </entry>
   <entry>
     <title>{{_('Top Rated Books')}}</title>
-    <link href="{{url_for('opds.feed_best_rated')}}" type="application/atom+xml;profile=opds-catalog"/>
+    <link href="{{url_for('opds.feed_best_rated')}}" type="application/atom+xml;profile=opds-catalog;kind=acquisition"/>
     <id>{{url_for('opds.feed_best_rated')}}</id>
```
Comment (OzzieIsaacs) [inline cps/templates/index.xml:26]: Calibre adds this entry to it's feed and many opds reader are more compatible to calibre than to the spec and some of them are very strict, if there is anything different form what they are expecting it fails. Last time I changed something several readers caused problems -> No change
Example:
  <entry>
    <title>Von Neue</title>
    <id>calibre-navcatalog:9302fc328c142a43b4f37efeb050ef65e9013b98</id>
    <updated>2020-02-01T08:48:56+00:00</updated>
    <content type="text">Bücher sortiert nach Datum</content>
    <link type="application/atom+xml;type=feed;profile=opds-catalog" href="/opds/navcatalog/4f6e6577657374?library_id=Ebooks"/>
  </entry>
It differs for current behavior, yes. If you give me an example of an non working reader I'm willing to check

File: cps/templates/index.xml [hunk 6]
```diff
@@ -15,101 +15,101 @@
...
     <updated>{{ current_time }}</updated>
     <content type="text">{{_('Popular publications from this catalog based on Rating.')}}</content>
   </entry>
   <entry>
     <title>{{_('Recently added Books')}}</title>
-    <link href="{{url_for('opds.feed_new')}}" type="application/atom+xml;profile=opds-catalog"/>
+    <link href="{{url_for('opds.feed_new')}}" type="application/atom+xml;profile=opds-catalog;kind=acquisition"/>
     <id>{{url_for('opds.feed_new')}}</id>
```
Comment (OzzieIsaacs) [inline cps/templates/index.xml:33]: Calibre left this entry to it's feed and many opds reader are more compatible to calibre than to the spec and some of them are very strict, if there is anything different form what they are expecting it fails. Last time I changed something several readers caused problems -> No change

File: cps/templates/index.xml [hunk 7]
```diff
@@ -15,101 +15,101 @@
...
     <updated>{{ current_time }}</updated>
     <content type="text">{{_('The latest Books')}}</content>
   </entry>
   <entry>
     <title>{{_('Random Books')}}</title>
-    <link href="{{url_for('opds.feed_discover')}}" type="application/atom+xml;profile=opds-catalog"/>
+    <link href="{{url_for('opds.feed_discover')}}" type="application/atom+xml;profile=opds-catalog;kind=acquisition"/>
     <id>{{url_for('opds.feed_discover')}}</id>
```
Comment (OzzieIsaacs) [inline cps/templates/index.xml:40]: Calibre left this entry to it's feed and many opds reader are more compatible to calibre than to the spec and some of them are very strict, if there is anything different form what they are expecting it fails. Last time I changed something several readers caused problems -> No change

File: cps/templates/index.xml [hunk 8]
```diff
@@ -15,101 +15,101 @@
...
     <updated>{{ current_time }}</updated>
     <content type="text">{{_('Show Random Books')}}</content>
   </entry>
   {% if not current_user.is_anonymous %}
   <entry>
     <title>{{_('Read Books')}}</title>
-    <link href="{{url_for('opds.feed_read_books')}}" type="application/atom+xml;profile=opds-catalog"/>
+    <link href="{{url_for('opds.feed_read_books')}}" type="application/atom+xml;profile=opds-catalog;kind=acquisition"/>
     <id>{{url_for('opds.feed_read_books')}}</id>
```
Comment (OzzieIsaacs) [inline cps/templates/index.xml:48]: Calibre left this entry to it's feed and many opds reader are more compatible to calibre than to the spec and some of them are very strict, if there is anything different form what they are expecting it fails. Last time I changed something several readers caused problems -> No change

File: cps/templates/index.xml [hunk 9]
```diff
@@ -15,101 +15,101 @@
...
     <updated>{{ current_time }}</updated>
-    <content type="text">{{_('Read Books')}}</content>
+    <content type="text">{{_('Show Read Books')}}</content>
   </entry>
   <entry>
     <title>{{_('Unread Books')}}</title>
-    <link href="{{url_for('opds.feed_unread_books')}}" type="application/atom+xml;profile=opds-catalog"/>
+    <link href="{{url_for('opds.feed_unread_books')}}" type="application/atom+xml;profile=opds-catalog;kind=acquisition"/>
     <id>{{url_for('opds.feed_unread_books')}}</id>
```
Comment (OzzieIsaacs) [inline cps/templates/index.xml:55]: Calibre left this entry to it's feed and many opds reader are more compatible to calibre than to the spec and some of them are very strict, if there is anything different form what they are expecting it fails. Last time I changed something several readers caused problems -> No change

File: cps/templates/index.xml [hunk 10]
```diff
@@ -15,101 +15,101 @@
...
     <updated>{{ current_time }}</updated>
-    <content type="text">{{_('Unread Books')}}</content>
+    <content type="text">{{_('Show Unread Books')}}</content>
   </entry>
   {% endif %}
   <entry>
     <title>{{_('Authors')}}</title>
-    <link href="{{url_for('opds.feed_authorindex')}}" type="application/atom+xml;profile=opds-catalog"/>
+    <link href="{{url_for('opds.feed_authorindex')}}" type="application/atom+xml;profile=opds-catalog;kind=acquisition"/>
     <id>{{url_for('opds.feed_authorindex')}}</id>
```
Comment (OzzieIsaacs) [inline cps/templates/index.xml:63]: Calibre left this entry to it's feed and many opds reader are more compatible to calibre than to the spec and some of them are very strict, if there is anything different form what they are expecting it fails. Last time I changed something several readers caused problems -> No change

File: cps/templates/index.xml [hunk 11]
```diff
@@ -15,101 +15,101 @@
...
     <updated>{{ current_time }}</updated>
     <content type="text">{{_('Books ordered by Author')}}</content>
   </entry>
   <entry>
     <title>{{_('Publishers')}}</title>
-    <link href="{{url_for('opds.feed_publisherindex')}}" type="application/atom+xml;profile=opds-catalog"/>
+    <link href="{{url_for('opds.feed_publisherindex')}}" type="application/atom+xml;profile=opds-catalog;kind=acquisition"/>
     <id>{{url_for('opds.feed_publisherindex')}}</id>
```
Comment (OzzieIsaacs) [inline cps/templates/index.xml:70]: Calibre left this entry to it's feed and many opds reader are more compatible to calibre than to the spec and some of them are very strict, if there is anything different form what they are expecting it fails. Last time I changed something several readers caused problems -> No change

File: cps/templates/index.xml [hunk 12]
```diff
@@ -15,101 +15,101 @@
...
     <updated>{{ current_time }}</updated>
     <content type="text">{{_('Books ordered by publisher')}}</content>
   </entry>
   <entry>
     <title>{{_('Categories')}}</title>
-    <link href="{{url_for('opds.feed_categoryindex')}}" type="application/atom+xml;profile=opds-catalog"/>
+    <link href="{{url_for('opds.feed_categoryindex')}}" type="application/atom+xml;profile=opds-catalog;kind=acquisition"/>
     <id>{{url_for('opds.feed_categoryindex')}}</id>
```
Comment (OzzieIsaacs) [inline cps/templates/index.xml:77]: Calibre left this entry to it's feed and many opds reader are more compatible to calibre than to the spec and some of them are very strict, if there is anything different form what they are expecting it fails. Last time I changed something several readers caused problems -> No change

File: cps/templates/index.xml [hunk 13]
```diff
@@ -15,101 +15,101 @@
...
     <updated>{{ current_time }}</updated>
     <content type="text">{{_('Books ordered by category')}}</content>
   </entry>
   <entry>
     <title>{{_('Series')}}</title>
-    <link href="{{url_for('opds.feed_seriesindex')}}" type="application/atom+xml;profile=opds-catalog"/>
+    <link href="{{url_for('opds.feed_seriesindex')}}" type="application/atom+xml;profile=opds-catalog;kind=acquisition"/>
     <id>{{url_for('opds.feed_seriesindex')}}</id>
```
Comment (OzzieIsaacs) [inline cps/templates/index.xml:84]: Calibre left this entry to it's feed and many opds reader are more compatible to calibre than to the spec and some of them are very strict, if there is anything different form what they are expecting it fails. Last time I changed something several readers caused problems -> No change

File: cps/templates/index.xml [hunk 14]
```diff
@@ -15,101 +15,101 @@
...
     <updated>{{ current_time }}</updated>
     <content type="text">{{_('Books ordered by series')}}</content>
   </entry>
   <entry>
     <title>{{_('Languages')}}</title>
-    <link href="{{url_for('opds.feed_languagesindex')}}" type="application/atom+xml;profile=opds-catalog"/>
+    <link href="{{url_for('opds.feed_languagesindex')}}" type="application/atom+xml;profile=opds-catalog;kind=acquisition"/>
     <id>{{url_for('opds.feed_languagesindex')}}</id>
```
Comment (OzzieIsaacs) [inline cps/templates/index.xml:91]: Calibre left this entry to it's feed and many opds reader are more compatible to calibre than to the spec and some of them are very strict, if there is anything different form what they are expecting it fails. Last time I changed something several readers caused problems -> No change

File: cps/templates/index.xml [hunk 15]
```diff
@@ -15,101 +15,101 @@
...
     <updated>{{ current_time }}</updated>
     <content type="text">{{_('Books ordered by Languages')}}</content>
   </entry>
   <entry>
     <title>{{_('Ratings')}}</title>
-    <link href="{{url_for('opds.feed_ratingindex')}}" type="application/atom+xml;profile=opds-catalog"/>
+    <link href="{{url_for('opds.feed_ratingindex')}}" type="application/atom+xml;profile=opds-catalog;kind=acquisition"/>
     <id>{{url_for('opds.feed_ratingindex')}}</id>
```
Comment (OzzieIsaacs) [inline cps/templates/index.xml:98]: Calibre left this entry to it's feed and many opds reader are more compatible to calibre than to the spec and some of them are very strict, if there is anything different form what they are expecting it fails. Last time I changed something several readers caused problems -> No change

File: cps/templates/index.xml [hunk 16]
```diff
@@ -15,101 +15,101 @@
...
     <updated>{{ current_time }}</updated>
     <content type="text">{{_('Books ordered by Rating')}}</content>
   </entry>
 
   <entry>
     <title>{{_('File formats')}}</title>
-    <link href="{{url_for('opds.feed_formatindex')}}" type="application/atom+xml;profile=opds-catalog"/>
+    <link href="{{url_for('opds.feed_formatindex')}}" type="application/atom+xml;profile=opds-catalog;kind=navigation"/>
     <id>{{url_for('opds.feed_formatindex')}}</id>
```
Comment (OzzieIsaacs) [inline cps/templates/index.xml:106]: Calibre left this entry to it's feed and many opds reader are more compatible to calibre than to the spec and some of them are very strict, if there is anything different form what they are expecting it fails. Last time I changed something several readers caused problems -> No change

File: cps/templates/index.xml [hunk 17]
```diff
@@ -15,101 +15,101 @@
...
     <updated>{{ current_time }}</updated>
     <content type="text">{{_('Books ordered by file formats')}}</content>
   </entry>
   <entry>
     <title>{{_('Shelves')}}</title>
-    <link  href="{{url_for('opds.feed_shelfindex')}}" type="application/atom+xml;profile=opds-catalog"/>
+    <link  href="{{url_for('opds.feed_shelfindex')}}" type="application/atom+xml;profile=opds-catalog;kind=navigation"/>
     <id>{{url_for('opds.feed_shelfindex')}}</id>
```
Comment (OzzieIsaacs) [inline cps/templates/index.xml:113]: Calibre left this entry to it's feed and many opds reader are more compatible to calibre than to the spec and some of them are very strict, if there is anything different form what they are expecting it fails. Last time I changed something several readers caused problems -> No change

File: cps/templates/index.xml [hunk 18]
```diff
@@ -15,101 +15,101 @@
...
   </entry>
   <entry>
     <title>{{_('Unread Books')}}</title>
-    <link href="{{url_for('opds.feed_unread_books')}}" type="application/atom+xml;profile=opds-catalog"/>
+    <link href="{{url_for('opds.feed_unread_books')}}" type="application/atom+xml;profile=opds-catalog;kind=acquisition"/>
     <id>{{url_for('opds.feed_unread_books')}}</id>
     <updated>{{ current_time }}</updated>
-    <content type="text">{{_('Unread Books')}}</content>
+    <content type="text">{{_('Show Unread Books')}}</content>
   </entry>
```
Comment (OzzieIsaacs) [inline cps/templates/index.xml:58]: The text is identical to the text in the "normal" user interface. There was recently a issue about different text between UI and opds feed. Furthermore all other text are also not saying showing like Publisher, Categories and so on -> no change

File: cps/templates/index.xml [hunk 19]
```diff
@@ -15,101 +15,101 @@
...
   {% if not current_user.is_anonymous %}
   <entry>
     <title>{{_('Read Books')}}</title>
-    <link href="{{url_for('opds.feed_read_books')}}" type="application/atom+xml;profile=opds-catalog"/>
+    <link href="{{url_for('opds.feed_read_books')}}" type="application/atom+xml;profile=opds-catalog;kind=acquisition"/>
     <id>{{url_for('opds.feed_read_books')}}</id>
     <updated>{{ current_time }}</updated>
-    <content type="text">{{_('Read Books')}}</content>
+    <content type="text">{{_('Show Read Books')}}</content>
   </entry>
```
Comment (OzzieIsaacs) [inline cps/templates/index.xml:51]: The text is identical to the text in the "normal" user interface. There was recently a issue about different text between UI and opds feed. Furthermore all other text are also not saying showing like Publisher, Categories and so on -> no change

File: cps/templates/feed.xml [hunk 20]
```diff
@@ -49,26 +49,32 @@
...
     {% if entry.publishers.__len__() > 0 %}
-      <publisher>
+      <dc:publisher>
         <name>{{entry.publishers[0].name}}</name>
-      </publisher>
+      </dc:publisher>
+    {% endif %}
+    {% if entry.pubdate %}
+      <dc:issued>{{entry.pubdate}}</dc:issued>
     {% endif %}
```
Comment (OzzieIsaacs) [inline cps/templates/feed.xml:58]: I'm not sure about this one: According to spec's example it should be 
<dc:issued>2009-10-01</dc:issued>
Calibre outputs the following:
    <updated>2013-11-30T14:17:10+00:00</updated>
    <published>2013-11-25T17:26:09+00:00</published>
    <dc:date>2012-12-05T23:00:00+00:00</dc:date>
In both cases I think the outputs needs to be formatted like for current time, and I think this was the reason I didn't do it. 
If there is no specific problem, I don't want to change it

File: cps/templates/feed.xml [hunk 21]
```diff
@@ -49,26 +49,32 @@
...
     {% for lang in entry.languages %}
-      <dcterms:language>{{lang.lang_code}}</dcterms:language>
+      <dc:language>{{lang.lang_code}}</dc:language>
     {% endfor %}
```
Comment (OzzieIsaacs) [inline cps/templates/feed.xml:61]: Calibre doesn't output it, according to spec dc: would be indeed correct

File: cps/templates/feed.xml [hunk 22]
```diff
@@ -49,26 +49,32 @@
       </author>
     {% endif %}
     {% if entry.publishers.__len__() > 0 %}
-      <publisher>
+      <dc:publisher>
         <name>{{entry.publishers[0].name}}</name>
-      </publisher>
+      </dc:publisher>
```
Comment (OzzieIsaacs) [inline cps/templates/feed.xml:54]: According to specification correct, calibre doesn't output the entry
-> better not changing unless really needed

**Comment by OzzieIsaacs** [comment]
In general I'm not convinced to change the feed, as I can test it against a mockup and against what is programmed, but in the past there have been several issues with reader which are excpecting it specially like calibre and not like the spec and optional and this is not testable. And as you see the code documentation I didn't make remarks which line is needad as it is and which one can be changed savely.

**Comment by xcffl** [comment]
> In general I'm not convinced to change the feed, as I can test it against a mockup and against what is programmed, but in the past there have been several issues with reader which are excpecting it specially like calibre and not like the spec and optional and this is not testable. And as you see the code documentation I didn't make remarks which line is needad as it is and which one can be changed savely.

Well actually I submit this PR because the software I use doesn't recognize this format. At least the namespace should be Atom because that's what it complains.

I'll look into details later.

**Comment by OzzieIsaacs** [comment]
Which software are you using?

**Comment by xcffl** [comment]
> Which software are you using?

Will... Google search console actually. It asks for strict spec compliance.

**Comment by OzzieIsaacs** [comment]
I merged the head dcterms -> dc change, I rejected the other changes

**Pull request merged**